### ISSUE TITLE
500 error when trying to create store on gaia storage

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-core#639]
<img width="760" alt="screen shot 2017-10-09 at 11 04 55 pm" src="https://user-images.githubusercontent.com/597182/31344804-4ec0e65a-ad46-11e7-8aca-e5047136a63e.png">

Tried this with http://blockstack-todos.appartisan.com

I'm using the virtual env from https://circleci.com/gh/blockstack/blockstack-core/799 

[api_endpoint.log.zip](https://github.com/blockstack/blockstack-core/files/1368936/api_endpoint.log.zip)

**Comment by larrysalibra** [comment]
I'm having this problem with the dropbox driver too:
[api_endpoint-dropbox.log.zip](https://github.com/blockstack/blockstack-core/files/1369023/api_endpoint-dropbox.log.zip)

**Comment by jcnelson** [comment]
The problem is that you use Mac OS ;) 

```
[2017-10-09 23:24:22,192] [ERROR] [rpc:4009] (10717.123145453711360) 'module' object has no attribute 'EREMOTEIO'
Traceback (most recent call last):
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.2-py2.7.egg/blockstack_client/rpc.py", line 4006, in _dispatch
    return route_method( session, path_info, *route_args )
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.2-py2.7.egg/blockstack_client/rpc.py", line 1380, in POST_store
    return self._store_signed_datastore(ses, path_info, request)
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.2-py2.7.egg/blockstack_client/rpc.py", line 1355, in _store_signed_datastore
    res = data.put_datastore_info( datastore_info, sigs_info, root_tombstones, config_path=self.server.config_path )
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.2-py2.7.egg/blockstack_client/data.py", line 1869, in put_datastore_info
    res = put_inode_data( datastore, datastore_info['root_blob_header'], datastore_sigs['root_sig'], datastore_info['root_blob_idata'], config_path=config_path, proxy=proxy)
  File "/Users/larry/git/blockstack-testing/master/blockstack-venv/lib/python2.7/site-packages/blockstack-0.17.0.2-py2.7.egg/blockstack_client/data.py", line 2671, in put_inode_data
    return {'error': 'Failed to replicate inode data to at least one driver', 'errno': errno.EREMOTEIO}
AttributeError: 'module' object has no attribute 'EREMOTEIO'
127.0.0.1 - - [09/Oct/2017 23:24:22] "POST /v1/stores HTTP/1.1" 500 -
```

`EREMOTEIO` isn't defined in `errno.h` on Macs for some reason, which means it won't be defined in Python either.

I'll make a patch that makes `EREMOTEIO` its own variable, instead of something we get from `errno`.  I'll also make sure Gaia v2 doesn't have this problem.

**Comment by jcnelson** [comment]
There are a couple of other things too.  Investigating...

**Comment by jcnelson** [comment]
Can you try again with the `hotfix/storage-errno` branch?  It doesn't fix the problem with replication that you're seeing, but it will make the above error path report the problem correctly.  It also tweaks the Gaia hub driver to log why it was misbehaving as well.

**Comment by larrysalibra** [comment]
@jcnelson attached log from the hotfix.
[api_endpoint-hotfix.log.zip](https://github.com/blockstack/blockstack-core/files/1369627/api_endpoint-hotfix.log.zip)

@shea256 is also having this same problem in our v0.17.0 as well.

**Comment by kantai** [comment]
Okay -- the EREMOTEIO error was causing the 500 response, however, as @jcnelson mentions, there's an underlying failure which causes the REMOTEIO error response (which is itself an error path).

The underlying failure has to do with the `client.ini` setting:

```
storage_drivers_required_write = disk,dropbox
```

That's the default setting in production, and the default setting in regtest is:
```
storage_drivers_required_write = disk
```

Switching production to `storage_drivers_required_write=disk` _should_ solve this issue, am trying it with that hotfix branch now.

**Comment by kantai** [comment]
Oh, there was also a problem with the CDN update to gaia_hub -- the URL it was responding with was wrong, which caused the replication error you saw.

**Comment by larrysalibra** [comment]
@jcnelson @kantai - it works great for me on regtest now! i'll try out the hotfix in a production build tomorrow and push a hotfix of the browser if it checks out. Thanks for the quick turnaround on this!

**Comment by kantai** [comment]
Great! I'll merge the hotfix branch into master today.

**Comment by kantai** [comment]
This was the fix shipped in browser v0.17.1
